shared-libs/cht-datasource/src/libs/error.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Represents an error that occurs when an invalid argument is provided.
+ * This error is typically thrown when a function or method receives an argument
+ * that doesn't meet the expected criteria or constraints.
+ */
+export class InvalidArgumentError extends Error {
+  /**
+   * Constructor
+   * @param message a descriptive error message why the error was raised
+   */
+  constructor(message: string) {
+    super(message);
+    this.name = message;
```

Comment:
```suggestion
    this.name = 'InvalidArgumentError';
```

I would expect the name property to hold the name of the class. The message can always be accessed in 
the message property since the `Error` constructor assigns it there.